webhook_xblock/webhook_xblock.py

```diff
@@ -47,21 +99,73 @@ def student_view(self, context=None):
         frag.initialize_js('WebhookXblock')
         return frag
 
-    # TO-DO: change this handler to perform your own actions.  You may need more
-    # than one handler, or you may not need any handlers at all.
+    def studio_view(self, context=None):
+        """
+        Creates a fragment used to display the edit view in the Studio.
+        This view allows the user (staff) to set the webhook configurations.
+        """
+        html_str = self.resource_string("static/html/webhook_xblock_edit.html")
+
+        frag = Fragment(html_str.format(
+            self=self, 
+            frequency=self.frequency, 
+            extra_info=json.dumps(self.extra_info).replace("\"", "&quot;"), 
+            send_course_grade=self.send_course_grade, 
+            send_async=self.send_async,
+            component_text=self.component_text,
+            button_text=self.button_text,
+            webhook_url=self.webhook_url or "",
+        ))
+        frag.add_css(self.resource_string("static/css/webhook_xblock.css"))
+
+        frag.add_javascript(self.resource_string("static/js/src/webhook_xblock_edit.js"))
+        frag.initialize_js('WebhookXblockEditBlock')
+        return frag
+
     @XBlock.json_handler
-    def increment_count(self, data, suffix=''):
+    def send_payload(self, data, suffix=''):
         """
-        An example handler, which increments the data.
+        Handler that triggers the webhook and sends the payload.
         """
-        # Just to show data coming in...
-        assert data['hello'] == 'world'
+        if not self.check_if_running_from_studio() and (self.frequency!='one-time' or self.mark_as_visited()):
+            current_anonymous_student_id = self.runtime.anonymous_student_id
+            student = self.runtime.get_real_user(current_anonymous_student_id)
+            serialized_student = AccountUserSerializer(student)
+
+            payload = {
+                "timestamp": datetime.datetime.now().isoformat(),
+                "anonymous_student_id": current_anonymous_student_id,
+                "student": serialized_student.data,
+                "course_id": str(self.runtime.course_id),
+            }
+            payload.update(self.extra_info)
+            payload = flatten_dict(payload)
 
-        self.count += 1
-        return {"count": self.count}
+            if self.send_async:
+                response = send_payload_async(kwargs={"data": payload})
```

Comment:
this is not async, you have to write something like  func.delay() or func.apply_async()